### PR TITLE
[Ballerina to OAS] Add hateoas support proposal to the docs

### PR DESCRIPTION
**Pull request by SachinAkash01**

### CODE REVIEW

File: docs/ballerina-to-oas/proposals/hateoas_support.md [hunk 1]
```diff
@@ -0,0 +1,164 @@
+# Proposal: Map Ballerina HATEOAS Links to OpenAPI Specification
+
+_Owners_: [@SachinAkash01](https://github.com/SachinAkash01)  
+_Reviewers_: [@lnash94](https://github.com/lnash94) [@TharmiganK](https://github.com/TharmiganK)
```
Comment (ayeshLK) [inline docs/ballerina-to-oas/proposals/hateoas_support.md]:
```suggestion
_Owners_: @SachinAkash01  
_Reviewers_: @lnash94 @TharmiganK    
```

**Review by ayeshLK — APPROVED** [review]
LGTM

File: docs/ballerina-to-oas/proposals/hateoas_support.md [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+## Description
+There are two ways of generating links between the resources in Ballerina, manually defining links between resources and
+automatically generating links between resources.  The feature implementation focuses on integrating Ballerina 
+Hypermedia constraints into generated OpenAPI specifications to accurately document the API.  In this proposal we will 
+be addressing the automatic linking between the resources using `ResourceConfig` annotation in Ballerina.
+
+Currently in manual linking approach, we are generating the response component schema by including `*http:Links`. 
+There will be no changes to this behaviour of generating the OpenAPI specification.
+
+**Sample Ballerina Code (Manually defining links between resources):**
```
Comment (ayeshLK) [inline docs/ballerina-to-oas/proposals/hateoas_support.md:39]: We are not doing this part right ?

Comment (SachinAkash01) [reply docs/ballerina-to-oas/proposals/hateoas_support.md:39]: Its the current way how the manual linking between resources is reflected in OAS I have mentioned that this current behavior will not be changed..

**Pull request merged**